src/saturn_engine/utils/error_handling.py

```diff
@@ -0,0 +1,148 @@
+import os
+import re
+import sys
+from dataclasses import dataclass
+from functools import lru_cache
+from textwrap import dedent
+
+from saturn_engine.worker.executors.bootstrap import RemoteException
+
+
+@dataclass
+class ExceptionDetails:
+    message: str
+    exception_type: Exception
+    module: str
+    lineno: int
+
+
+@dataclass
+class ExceptionFilter(ExceptionDetails):
+    action: str
+
+
+def resolve_exception(exception_type: str) -> Exception:
```

Comment:
I would use our existing `import_name` instead, this ensure the expected behavior match all other place we degine class names.